Common/Core/RecoDecay.h

```diff
@@ -913,6 +913,71 @@ class RecoDecay
     return true;
   }
 
+  /// Finds the origin (from charm hadronisation or beauty-hadron decay) of charm hadrons.
+  /// \param particlesMC  table with MC particles
+  /// \param particle  MC particle
+  /// \param searchUpToQuark if true tag origin based on charm/beauty quark otherwise on b-hadron
+  /// \return an integer corresponding to the origin (0: none, 1: prompt, 2: nonprompt) as in hf_cand::OriginType (see PWGHF/DataModel/HFSecondaryVertex.h)
+  template <typename T>
+  static int checkCharmHadronOrigin(const T& particlesMC,
```

Comment:
Maybe a better name would be `getCharmHadronOrigin` since the result is directly returned.

Reply:
Done!